src/main/kotlin/org/opensearch/commons/alerting/model/BaseAlert.kt

```diff
@@ -0,0 +1,208 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.common.lucene.uid.Versions
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.NO_SCHEMA_VERSION
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.authuser.User
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.common.io.stream.Writeable
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.XContentBuilder
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.core.xcontent.XContentParserUtils
+import java.io.IOException
+import java.time.Instant
+
+/** CorrelationAlert and Alert can extend the UnifiedAlert class to inherit the common fields and behavior
+ * of UnifiedAlert class.
+ */
+open class BaseAlert(
+    open val id: String = Alert.NO_ID,
+    open val version: Long = Alert.NO_VERSION,
+    open val schemaVersion: Int = NO_SCHEMA_VERSION,
+    open val user: User?,
+    open val triggerName: String,
+
+    // State will be later moved to this Class  (after `monitorBasedAlerts` extend this Class)
+    open val state: Alert.State,
+    open val startTime: Instant,
+    open val endTime: Instant? = null,
+    open val acknowledgedTime: Instant? = null,
+    open val errorMessage: String? = null,
+    open val severity: String,
+    open val actionExecutionResults: List<ActionExecutionResult>
+) : Writeable, ToXContent {
+
+    init {
+        if (errorMessage != null) {
+            require((state == Alert.State.DELETED) || (state == Alert.State.ERROR) || (state == Alert.State.AUDIT)) {
+                "Attempt to create an alert with an error in state: $state"
+            }
+        }
+    }
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
```

Comment:
are there writeable tests for these constructors? like this. https://github.com/opensearch-project/common-utils/blob/main/src/test/kotlin/org/opensearch/commons/alerting/model/WriteableTests.kt

Reply:
no, but here src/test/kotlin/org/opensearch/commons/alerting/CorrelationAlertTests.kt in `test correlation parse function` we are validating such scenarios